u8timeseries/models/torch_forecasting_model.py

```diff
@@ -0,0 +1,505 @@
+        """ Pytorch-based Forecasting Model.
+
+        This class is meant to be inherited to create a new pytorch-based forecasting module.
+        When subclassing this class, please make sure to set the self.model attribute
+        in the __init__ function and then call super().__init__ while passing the kwargs.
+
+        Parameters
+        ----------
+        output_length
+            Number of time steps to be output by the forecasting module.
+        input_length
+            Number of past time steps that are fed to the forecasting module.
+        batch_size
+            Number of time series (input and output sequences) used in each training pass.
+        n_epochs
+            Number of epochs over which to train the model.
+        optimizer_cls
+            The PyTorch optimizer class to be used (default: `torch.optim.Adam`).
+        optimizer_kwargs
+            Optionally, some keyword arguments for the PyTorch optimizer (e.g., `{'lr': 1e-3}`)
+            for specifying a learning rate. Otherwise the default values of the selected `optimizer_cls`
+            will be used.
+        lr_scheduler_cls
+            Optionally, the PyTorch learning rate scheduler class to be used. Specifying `None` corresponds
+            to using a constant learning rate.
+        lr_scheduler_kwargs
+            Optionally, some keyword arguments for the PyTorch optimizer.
+        loss_fn
+            PyTorch loss function used for training (default: `torch.nn.MSELoss()`).
+        model_name
+            Name of the model. Used for creating the checkpoints and saving tensorboard data.
+        work_dir
+            Path of the working directory, where to save checkpoints and Tensorboard summaries.
+            (default: current working directory).
+        log_tensorboard
+            If set, use Tensorboard to log the different parameters. The logs will be located in:
+            `[work_dir]/.u8timeseries/runs/`.
+        nr_epochs_val_period
+            Number of epochs to wait before evaluating the validation loss (if a validation
+            `TimeSeries` is passed to the `fit()` method).
+        torch_device_str
+            Optionally, a string indicating the torch device to use. (default: "cuda:0" if a GPU
+            is available, otherwise "cpu")
+        """
+
+        super().__init__()
+
+        raise_if_not(isinstance(self.model, nn.Module), 'Please make sure that self.model is set to a valid '
+                     'nn.Module subclass when subclassing TorchForecastingModel',
+                     logger)
+
+        if torch_device_str is None:
+            self.device = self._get_best_torch_device()
+        else:
+            self.device = torch.device(torch_device_str)
+
+        self.input_size = 1  # We support only univariate time series currently
+        self.input_length = input_length
+        self.output_length = output_length
+        self.seq_len = input_length
```

Comment:
Could we remove this one and keep only `self.input_length` ?

Reply:
done